packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts

```diff
@@ -143,11 +153,27 @@ export class SagemakerTrainTask implements ec2.IConnectable, sfn.IStepFunctionsT
         });
 
         // add the security groups to the connections object
-        if (this.props.vpcConfig) {
-            this.props.vpcConfig.securityGroups.forEach(sg => this.connections.addSecurityGroup(sg));
+        if (props.vpcConfig) {
```

Comment:
If this branch is not taken, `this.connections` is never assigned. Do you want to make it optional, or initialize with an empty object?

Reply:
Now initializes with empty connections object